src/types.ts

```diff
@@ -559,12 +572,11 @@ export type Router<TRouteDefCollection extends { [routeName: string]: any }> = {
 
   session: RouterSession<TRouteDefCollection>;
 
-  /**
-   * Registers a listener that is called when navigation occurs.
-   * Returns a function to remove the navigation listener.
-   */
-  listen(handler: NavigationHandler<TRouteDefCollection>): Unlisten;
+  useRoute: () => RouteDefCollectionRoute<TRouteDefCollection>;
+
+  Router: (props: { children?: any }) => any;
```

Comment:
Same here regarding children (`React.ReactNode`) + maybe `ReactElement<any, any>` for return type.

Reply:
Or maybe just `React.Component`?

Reply:
Same thoughts on this case as explained here: https://github.com/typehero/type-route/pull/39/files#r445521454